doc/source/user_guide/pyarrow.rst

```diff
@@ -0,0 +1,162 @@
+the pyarrow array constructor on the :class:`Series` or :class:`Index`.
+
+.. ipython:: python
+
+   ser = pd.Series([1, 2, None], dtype="uint8[pyarrow]")
+   pa.array(ser)
+
+   idx = pd.Index(ser)
+   pa.array(idx)
+
+Operations
+----------
+
+PyArrow data structure integration is implemented through pandas' :class:`~pandas.api.extensions.ExtensionArray` :ref:`interface <extending.extension-types>`;
+therefore, supported functionality exists where this interface is integrated within the pandas API. Additionally, this functionality
+is accelerated with PyArrow `compute functions <https://arrow.apache.org/docs/python/api/compute.html>`__ where available. This includes:
+
+* Numeric aggregations
+* Numeric arithmetic
+* Numeric rounding
+* Logical and comparison functions
+* String functionality
+* Datetime functionality
+
+The following are just some examples of operations that are accelerated by native PyArrow compute functions.
+
+.. ipython:: python
+
+   ser = pd.Series([-1.545, 0.211, None], dtype="float32[pyarrow]")
+   ser.mean()
+   ser + ser
+   ser > (ser + 1)
+
+   ser.dropna()
+   ser.isna()
+   ser.fillna(0)
+
+   ser_str = pd.Series(["a", "b", None], dtype="string[pyarrow]")
+   ser_str.str.startswith("a")
+
+   from datetime import datetime
+   pa_type = pd.ArrowDtype(pa.timestamp("ns"))
+   ser_dt = pd.Series([datetime(2022, 1, 1), None], dtype=pa_type)
+   ser_dt.dt.strftime("%Y-%m")
+
+I/O Reading
+-----------
+
+PyArrow also provides IO reading functionality that has been integrated into several pandas IO readers. The following
+functions provide an ``engine`` keyword that can dispatch to PyArrow to accelerate reading from an IO source.
+
+* :func:`read_csv`
+* :func:`read_json`
+* :func:`read_orc`
+* :func:`read_feather`
+
+.. ipython:: python
+
+   import io
+   data = io.StringIO("""a,b,c
+      1,2.5,True
+      3,4.5,False
+   """)
+   df = pd.read_csv(data, engine="pyarrow")
+   df
+
+By default, these functions and all other IO reader functions return NumPy-backed data. These readers can return
+PyArrow-backed data by specifying the parameter ``use_nullable_dtypes=True`` **and** the global configuration option ``"mode.dtype_backend"``
+set to ``"pyarrow"``. A reader does not need to set ``engine="pyarrow"`` to necessarily return PyArrow-backed data.
+
+.. ipython:: python
+
+    import io
+    data = io.StringIO("""a,b,c,d,e,f,g,h,i
+        1,2.5,True,a,,,,,
+        3,4.5,False,b,6,7.5,True,a,
+    """)
+    with pd.option_context("mode.dtype_backend", "pyarrow"):
+        df_pyarrow = pd.read_csv(data, use_nullable_dtypes=True)
+    df_pyarrow.dtypes
+
+To simplify specifying ``use_nullable_dtypes=True`` in several functions, you can set a global option ``nullable_dtypes``
+to ``True``. You will still need to set the global configuration option ``"mode.dtype_backend"`` to ``pyarrow``.
+
+.. ipython:: python
+
+    pd.set_option("mode.dtype_backend", "pyarrow")
+    pd.options.mode.nullable_dtypes = True
```

Comment:
I think this has side effect, I guess we need to reset to False after this chapter is over (ci failure seems to be caused by this)

Reply:
Will convert to a code block instead

Reply:
That works as well, thx